### DISCUSSION TITLE
Options for SMC samplers

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1544] [New methods]
Allow a variety of options in SMC (potentially population MCMC too):

- Any sort of MCMC kernel (for example, adaptive MCMC, HMC) can be used,
- A range of possible resampling strategies (see [here](https://www.stats.ox.ac.uk/~doucet/doucet_johansen_tutorialPF2011.pdf) for a nice tutorial on these).

**Comment by MichaelClerx** [comment]
~See #465~

See #749
